### DISCUSSION TITLE
Returning Scalar Objects with Joins

### DISCUSSION THREAD

**Discussion by RPeterson0487** [sqlalchemy/sqlalchemy#11229] [Usage Questions] [answered]
Hello again,
This is my second of two questions.
I have this code (slight difference from my other question):
```py
cost = database_obj.cost
    inventory = database_obj.inventory
    
    with database_obj.session_factory() as session:
        test = session.scalars(
            sql.select(
                cost.Formula,
                cost.Lab_Description,
                sql.func.sum(inventory.Weight),
                sql.func.count("*"),
                inventory.Costs,
                inventory.Purchased
            )
            .join(
                # cost,
                inventory,
                cost.InkCode == inventory.InkType
            )
            .where(
                inventory.Deleted == "N"
            )
            .group_by(
                inventory.Purchased,
                cost.Formula,
                inventory.Costs.desc()
            )
        )
    
    for i in test:
        print(i)
```
When I run the code this way, I get the right number of results, however it is a list of strings that is only the first column (cost.Formula) and nothing else.  I've tried a few ways, including `session.execute(code).all()` but this gives a list that I must unpack.  Is there a better way to do this?

Thank you!
-- Rick

**Comment by CaselIT** [comment, marked as answer]
Hi,

Not really. You are selecting multiple columns so you will get back a tuple like object.
You can have back a mapping like object using `execute(...).mappings().all()`.

If the question is to only get the first row, there is `execute(..).first()` or `one()`.

You should take a look for the various options in the docs https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result

**Reply by RPeterson0487** [reply]
That gets me in the right direction.  Thank you so much.

**Reply by RPeterson0487** [reply]
And I will take a look at that page.
